packages/features/settings/layouts/SettingsLayout.tsx

```diff
@@ -105,14 +107,16 @@ const adminRequiredKeys = ["admin"];
 
 const useTabs = () => {
   const session = useSession();
+  const { data: user } = useMeQuery();
+  const { data: avatar } = useAvatarQuery();
 
   const isAdmin = session.data?.user.role === UserPermissionRole.ADMIN;
 
   tabs.map((tab) => {
-    if (tab.name === "my_account") {
-      tab.name = session.data?.user?.name || "my_account";
+    if (tab.href === "/settings/my-account") {
+      tab.name = user?.name || "my_account";
```

Comment:
although this works but we should look into how can we make sure useSession returns the latest data.

Reply:
@Udit-takkar ,

I used href because when the user changed the name in the profile settings, `tab.name` wouldn't remain the same as `my-account` in re-render.  

Here in [trpc].ts file, cacheRules is mentioned as `no-cache` for the session. So, it does revalidate each time but somehow it didn't update.

```
const cacheRules = {
      "viewer.public.session": `no-cache`,
      "viewer.public.i18n": `no-cache`,
      // Revalidation time here should be 1 second, per https://github.com/calcom/cal.com/pull/6823#issuecomment-1423215321
      "viewer.public.slots.getSchedule": `no-cache`, // FIXME
      "viewer.public.cityTimezones": `max-age=${ONE_DAY_IN_SECONDS}, stale-while-revalidate`,
    } as const;
```

Reply:
@Udit-takkar ,
Fixed. Thanks.